### PR TITLE
Patch relay-compiler to keep queryMap fresh

### PR DESCRIPTION
**Pull request by ds300**
Turns out we need the query map to be in-sync with the `__generated__` dir at all times because we use the query text from the query map at dev time to send requests to metaphysics.

Alas in recent versions relay-compiler makes the assumption that you either check both your query map and your `__generated__` dir into git, or neither. We only check in `__generated__` right now, which means that the two can become out-of-sync when you check out a different branch or pull changes from a remote. In this case relay will not update the queryMap properly. 

So we have two options:

1. No longer check in the `__generated__` dir. This seems to be the normal way to do relay.
2. Patch relay-compiler so it behaves like it used to, keeping everything in-sync all the time.

Here I've taken the latter approach for now, since it's less drastic. There's an active PR in relay-compiler (facebook/relay#3040) to add an option that we could eventually use for this, but for now it's a simple one-line patch.

I've also fixed an issue with our node_modules cache key which wasn't taking patch files into account

#trivial

### CODE REVIEW

File: node_modules.hash [hunk 1]
```diff
@@ -0,0 +1 @@
+1ed42055588af740d6b896f5da2a8e63
```
[the remark this answers is not in the document]
Comment (ds300) [reply node_modules.hash]: Only for CI

[the remark this answers is not in the document]
Comment (ds300) [reply node_modules.hash]: Oooops, I replied to this on the new github app and I guess the visual context there is unfamiliar so I didn't realise that this comment was telling me that I'd accidentally checked in the hash file 😬. No biggie, will remove it in an upcoming PR.

[the remark this answers is not in the document]
Comment (ashfurrow) [reply node_modules.hash]: Ooooh how do you find the app? I have some friends who work on it, happy to forward any feedback!

[the remark this answers is not in the document]
Comment (ds300) [reply node_modules.hash]: I like it a lot more than the mobile web, but it's still lacking some really basic stuff, e.g. can't see links to ci failures, can't see my closed PRs in my PRs list, ... great start though. The UI design is 👌

**Pull request merged**